### DISCUSSION TITLE
官方提供的数码管识别模型问题

### DISCUSSION THREAD

**Discussion by dly55555** [PaddlePaddle/PaddleOCR#13869] [Q&A] [answered]
win10
paddleocr版本:2.6.0
paddle_interface版本:2.2.2
从这个链接下载的模型:https://paddlepaddle.github.io/PaddleOCR/applications/overview.html,
下载之后用tools/export_model.py转为interface模型,然后使用cpp_infer进行测试,在识别的时候会崩溃,用vs调试会显示崩在paddle_interface里:0x00007FFF9487C023 (paddle_inference.dll)处(位于 ppocr.exe 中)引发的异常: 0xC0000005: 读取位置 0x0000000000000000 时发生访问冲突
转换interface模型使用的命令行:python tools/export_model.py -c m/config.yml -o Global.pretrained_model=./m/best_accuracy Global.load_static_weights=False Global.save_inference_dir=./inference/
生成了student和teacher两种模型,两种都试了,是一样的结果
有没有大神救救

**Comment by GreatV** [comment]
paddle_interface 版本换换试试

**Reply by dly55555** [reply]
2.3.2的预测库是一样的问题
2.1.1和2.0.1会报
![image](https://github.com/user-attachments/assets/2410afa2-769e-44b8-bcc8-1c5d247f115d)

**Reply by GreatV** [reply]
先试python端的，看看行不行

**Reply by dly55555** [reply]
python端结果:
![image](https://github.com/user-attachments/assets/f54489dc-70e1-4f04-827c-a6ef95069049)

**Reply by GreatV** [reply]
用的是paddle哪个版本呀

**Reply by dly55555** [reply]
paddleocr版本:2.6.0

**Reply by GreatV** [reply]
paddle框架的版本

**Reply by GreatV** [reply]
试试paddle2.5 https://www.paddlepaddle.org.cn/install/old?docurl=undefined

**Reply by GreatV** [reply]
估计是paddle框架和paddleocr版本不兼容

**Reply by dly55555** [reply]
这个脚本也需要依赖库吗,c++版本的库能下载的都试过了,包括:2.3.2,   2.2.2,    2.1.1,     2.0.1

**Reply by dly55555** [reply]
![image](https://github.com/user-attachments/assets/a1f9ecfe-0e46-444e-a734-73b4c4d1af28)

**Reply by GreatV** [reply]
因为这个是以前的模型，新版的不支持，得回退到以前的版本。

**Reply by GreatV** [reply]
这里有相同的问题 https://github.com/PaddlePaddle/Paddle/issues/46377

**Reply by dly55555** [reply]
使用:python tools/infer_rec.py -c m/config.yml -o Global.pretrained_model=./m/best_accuracy  Global.infer_img=t.png Global.use_gpu=False就可以了,在后面加了Global.use_gpu=False,
所以c++版本是不能用这个模型吗,需要自己用新版的ppocr训练吗

**Reply by GreatV** [reply, marked as answer]
c++版本可以用，要选python端能运行的 paddle 版本对应的paddle inference版本。但是要注意c++版本只支持crnn和db模型。

**Comment by dly55555** [comment]
问题解决,说一下过程:
问题核心是在转换interface模型时使用的python版的paddle_interface版本和c++的paddle_interface不一致,目前可直接下载的c++最新版本是2.3.2,所以在转换模型时把python板的paddle_interface版本降为2.3.2(与c++版本保持一致)就可以了
直接贴出命令行:python -m pip install paddlepaddle==2.3.2 -i https://pypi.tuna.tsinghua.edu.cn/simple
简单来说就是运行上面的命令之后,再转换interface模型就可以了
